src/Command/PqCommands/ContentHubPqNonTranslatables.php

```diff
@@ -0,0 +1,166 @@
+
+use Acquia\Console\ContentHub\Command\Helpers\DrupalServiceFactory;
+use Drupal\Core\Config\Entity\ConfigEntityInterface;
+use Drupal\Core\Entity\EntityTypeBundleInfoInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+
+/**
+ * Runs check against non-translatable entities.
+ */
+class ContentHubPqNonTranslatables extends ContentHubPqCommandBase {
+
+  /**
+   * The Drupal service factory service.
+   *
+   * @var \Acquia\Console\ContentHub\Command\Helpers\DrupalServiceFactory
+   */
+  protected $serviceFactory;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $defaultName = 'ach:pq:non-translatables';
+
+  /**
+   * Constructs a new ContentHubPqNonTranslatables object.
+   *
+   * @param \Acquia\Console\ContentHub\Command\Helpers\DrupalServiceFactory $drupalServiceFactory
+   *   The Drupal service factory service.
+   * @param string|null $name
+   *   The name of this command.
+   *
+   * @throws \Exception
+   */
+  public function __construct(DrupalServiceFactory $drupalServiceFactory, string $name = NULL) {
+    parent::__construct($name);
+    $this->serviceFactory = $drupalServiceFactory;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function configure() {
+    parent::configure();
+    $this->setDescription('Analyzes non-translatable entities and bundles.');
+    $this->setAliases(['ach-pq-nt']);
+    $this->addOption('entity-type', 'e', InputOption::VALUE_OPTIONAL, 'Run checks for the provided entity type. Example node,user,paragraph', 'node');
+    $this->addUsage('ach:pq:non-translatables --entity-type "node,paragraph"');
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @throws \Exception
+   */
+  protected function runCommand(InputInterface $input, PqCommandResult $result): int {
+    $entityTypeOption = $input->getOption('entity-type');
+    /** @var \Drupal\Core\Entity\EntityTypeManagerInterface $etm */
+    $etm = $this->serviceFactory->getDrupalService('entity_type.manager');
+    /** @var \Drupal\Core\Entity\EntityTypeBundleInfoInterface $bundle_info */
+    $bundle_info = $this->serviceFactory->getDrupalService('entity_type.bundle.info');
+    $entityTypes = explode(',', $entityTypeOption);
+    $entity_type_definitions = $etm->getDefinitions();
+    foreach ($entityTypes as $entityType) {
+      $entityTypeLabel = $this->getEntityTypeLabel($entityType, $etm);
+      $bundles = $this->getBundles($entityType, $bundle_info);
+      $kriName = 'Entity Type: ' . $entityTypeLabel;
+      if (empty($bundles) || $entity_type_definitions[$entityType]->entityClassImplements(ConfigEntityInterface::class)) {
+        $result->setIndicator(
+          $kriName,
+          '',
+          'No non-translatable entities detected.'
+        );
+        return 0;
+      }
+      $entities = $this->loadEntities($etm, $entityType);
+      $nonTranslatables = [];
+      foreach ($entities as $entity) {
```

Comment:
Wondering if we really need to go through all entities to get whether an entity is translatable or not. I'd be interested in the fact whether a bundle of a given entity type is translatable or not.
If I have a content entity type and has bundle, which has a language_content_setting assigned to it, and content_translation is enabled there, we know that the bundle is translatable therefore the entity itself as well.
If it doesn't have a language_content_setting, or it is disabled it means the entity type is non-translatable.

Reply:
fixed in [6f47d59](https://github.com/acquia/contenthub-console/pull/35/commits/6f47d59a503068b9c275aabd01d10f35820bec72)